[MANX-274][dotnet] Remove unused code from MergedDateTimeParser

Bill7zz opened the pull request:
- Remove unused if statement from Japanese `MergeDateTimeParser` class

Bill7zz committed:
Update Japanese MergeDateTimeParser
- Remove unused if statement

Bill7zz committed:
Update Japanese MergeDateTimeParser
- Remove unused if statement

gasper-az reviewed (approved):
LGTM

JuanAr commented on .NET/Microsoft.Recognizers.Text.DateTime/Models/DateTimeModel.cs:
```diff
@@ -32,33 +32,24 @@ public List<ModelResult> Parse(string query, DateObject refTime)
-            }
-            catch (Exception)
-            {
-                // Nothing to do. Exceptions in parse should not break users of recognizers.
```
I'm a bit concerned about removing a try catch IF the Parser can throw an exception.

JuanAr replied:
I think, the PR will be approved maintaining the try/catch structure, removing the unused code if any.
Add a new task to review this specific smelling code: _Try/Catch in code without any explicit action_
Also, add a new issue in our repo to review. And then create the issue in MS repo to socialize and receive feedback and or approval to tackle it.

JuanAr reviewed (changes requested):
Feedback regarding some functionality changes

JuanAr committed:
[JavaScript] Rename DinoComparer to StringComparer and update implementation  (#1404)

* Rename DinoComparer to StringComparer and update implementation

JuanAr committed:
[CI][DotNet] Add Code Analysis as part of build process (#1405)

* Update install FxCopAnalyzers NuGet package in all projects of the solution

* Add code analysis rules in the Recognizers-Text.ruleset

* Add rule CA1021 (avoid out parameters) to Code Analysis ruleset

gasper-az committed:
Fixed rangeConnectorRegex Pattern to public static (#1418)

JuanAr committed:
[StyleCop] Fix warnings in Text.Number Project (#1419)

* Fix StyleCop warnings in SpanishNumberRangeParserConfiguration
-Move constructor before properties
-Move constructor initializers on their own line
-Add class access modifier

* Fix StyleCop warnings in Text.Number project
-XML comment analysis warning

* Fix StyleCop warning in NumberMode
-Update comment to fix CS1570 XML comment on 'variable' has badly formed

* Fix StyleCop warnings in BasePercentageExtractor
-Reorder properties an fields
-Set field to property to keep the variable name with upper-case
-Add retuns comments in methods
-Add trailing coma in multi-line initializers
-Use String.Empty instead empty strings

gasper-az committed:
[StyleCop]  Add warning fixes generating Breaking Changes  (#1420)

* Fix all StyleCop's warnings in TextMatchers files

- Reorder properties, fields and methods
- Remove trailing whitespace
- Add trailing comma
- Reformat setters and getters

* Fix all the StyleCop warnings in Parsers

Files:
    -BaseMergedUnitParser.cs
    -INumberWithUnitParserConfiguration.cs
Warnings:
    -SA1400	Class should declare an access modifier
    -SA1401	Field should be private
    -SA1122	Use string.Empty for empty strings

* Fix all the StyleCop warnings in NumberWithUnitParser

Warnings:
    -SA1401	 Field should be private
    -SA1122	 Use string.Empty for empty strings
    -SA1124	Do not use regions
    -Add file with the class UnitValue

gasper-az committed:
[Java] Rename DinoComparer to StringComparer and update implementation (#1422)

* Refactor DinoComparer
- Change name to StringComparer
- Polish implementation to make it more expressive

gasper-az committed:
Move EnglishDatetimeUtilityConfiguration class to utilities folder (#1423)

JuanAr reviewed (approved):
👍

gasper-az committed:
[DotNet][DateTime] Rename YearAfterRegex as DateAfterRegex (#1417)

* Renamed YearAfterRegex to DateAfterRegex.

* Include of the modified DateTimeDefinitions files

Irrelevances committed:
Support case “next” (#1400)

* Support relative references like "last", "next one",  "previous one"

* Add specs

* Regen python.

* Regen JavaScript.

* Support relative references like "last", "next one", "previous one"

* Support "the second to last", "the one before the last one", etc.

* Add base class.

* Revise base class.

* Support case "next".

* Optimize the regex .

* Add a EnablePreview option.

* Fix bug.

* Fix a bug.

Sothan committed:
Support WeekDayAndDay in DateTime model (#1416)

* Support WeekDayAndDay, e.g., Monday 21

* Add a few comments and minor fix

* Fixes according to the comments

JuanAr committed:
Fix IDateTimeExtractor type to IDateExtractor in English and Spanish languages (#1432)

JuanAr committed:
Refactor nested if-else, changed to a switch-case (#1433)

acblacktea committed:
Fix Spanish DatePeriod cases (#1438)

* Fix Spanish DatePeriod cases

* fix Spanish DatePeriod case and modify the error cases to right answer

iMicknl committed:
[.NET DateTime] Enable NL DateParser tests + remove 2 tests (#1440)

* Enable DateParser + remove 2 tests

* Mark 2 cases as NotSupported for now

* Remove duplicate case

guom08 committed:
Filter out the wrong extraction for "may differ" (#1435)

* Filter out "may differ" and update resources

guom08 committed:
Support combination of modifiers (#1439)

* Enable combination of modifiers

* Fix the parsing of "modifier + daterange"

JuanAr committed:
[DotNet][DateTime] Fix extractors and parsers name in Japanese and Chinese languages (#1431)

* Fix extractors and parsers class name in Chinese language

* Fix extractors and parsers class name in Japanese language

guom08 committed:
Correct behaviour for "upcoming/past + weekday" (#1437)

* Extend behaviour

* Rename and reorganize regexes

* Adjust the behavior of "following + weekday"

tellarin committed:
Removing unnecessary properties from SimpleConsole project. (#1441)

Bill7zz committed:
Merge branch 'fix/remove-unused-code-from-several-classes' of https://github.com/southworkscom/Recognizers-Text into fix/remove-unused-code-from-several-classes
